iac-extensions/docker/src/test/java/org/sonar/iac/docker/reports/HadolintImporterTest.java

```diff
@@ -154,6 +172,37 @@ void unknownRule(String reportPath) {
     verifyNoInteractions(mockAnalysisWarnings);
   }
 
+  @ParameterizedTest
+  @CsvSource({
+    "src/test/resources/hadolint/jsonFormat/unknownWarningRuleWithValidHadolintName.json, SC9999",
+    "src/test/resources/hadolint/sonarqubeFormat/unknownWarningRuleWithValidHadolintName.json, DL9999"})
+  void unknownWarningRuleWithValidHadolintFormat(String reportPath, String ruleId) {
+    File reportFile = new File(reportPath);
+    importReport(reportFile);
+    assertThat(context.allExternalIssues()).hasSize(1);
+    ExternalIssue issue = context.allExternalIssues().iterator().next();
+    assertThat(issue.ruleId()).isEqualTo(ruleId);
+    assertThat(issue.severity()).isEqualTo(Severity.MAJOR);
+    assertThat(issue.type()).isEqualTo(RuleType.CODE_SMELL);
+
+    verifyNoInteractions(mockAnalysisWarnings);
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "src/test/resources/hadolint/jsonFormat/unknownErrorRuleWithValidHadolintName.json, SC9999",
+    "src/test/resources/hadolint/sonarqubeFormat/unknownErrorRuleWithValidHadolintName.json, DL9999"})
+  void unknownErrorRuleWithValidHadolintFormat(String reportPath, String ruleId) {
+    File reportFile = new File(reportPath);
+    importReport(reportFile);
+    assertThat(context.allExternalIssues()).hasSize(1);
+    ExternalIssue issue = context.allExternalIssues().iterator().next();
+    assertThat(issue.ruleId()).isEqualTo(ruleId);
+    assertThat(issue.severity()).isEqualTo(Severity.CRITICAL);
+    assertThat(issue.type()).isEqualTo(RuleType.BUG);
```

Comment:
Maybe it's time for a custom assertion as we may use this for tflint and cfn-lint as well?

Reply:
Okay, i will open a ticket